### PR TITLE
Added gate changes support with associated tests

### PR DESCRIPTION
**Pull request by zack-rma**
Added gate changes support with associated test cases.

### CODE REVIEW

**Review by Enovotny — CHANGES_REQUESTED** [review]
please add gate_changes to _init file

**Review by Enovotny — APPROVED** [review]

**Review by Enovotny — APPROVED** [review]

File: cwms/locations/gate_changes.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+        {
+          "type": "gate-change",
+          "project-id": {
+            "office-id": "SPK",
+            "name": "BIGH"
+          },
+          "change-date": 1704096000000,
+          "pool-elevation": 3.0,
+          "protected": true,
+          "discharge-computation-type": {
```
Comment (adamkorynta) [inline cwms/locations/gate_changes.py:85]: Is the type itself stored, or just the display value?

**Review by adamkorynta — CHANGES_REQUESTED** [review]
some minor comments

File: cwms/locations/gate_changes.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    end: datetime
+        The end of the time range
+    start_time_inclusive: boolean, optional
+        Whether the returned gate changes should include data from the specified start timestamp. Default is `True`.
+    end_time_inclusive: boolean, optional
+        Whether the returned gate changes should include data from the specified end timestamp. Default is `False`.
+    unit_system: string, optional
+        The unit system to use for the gate changes. Can be SI (International Scientific) or EN (Imperial.)
+        Default is `EN`.
+    page_size: integer, optional
```
[the remark this answers is not in the document]
Comment (zack-rma) [reply cwms/locations/gate_changes.py:43]: Updated doc with language from CDA controller

**Review by adamkorynta — APPROVED** [review]

**Pull request merged**